### PR TITLE
CB-16073 - Freeipa scaling api rename

### PR DESCRIPTION
**Pull request by pdarvasi**
`FormFactor` --> `AvailabilityType` rename

### CODE REVIEW

File: freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/common/AvailabilityType.java [hunk 1]
```diff
@@ -6,17 +6,17 @@
...
 
-public enum FormFactor {
+public enum AvailabilityType {
 
     HA(3),
     TWO_NODE_BASED(2),
     NON_HA(1);
 
-    private static final Map<Integer, FormFactor> FORM_FACTOR_MAP_BY_INSTANCE_COUNT;
+    private static final Map<Integer, AvailabilityType> FORM_FACTOR_MAP_BY_INSTANCE_COUNT;
```
Comment (gergopapi2) [inline freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/common/AvailabilityType.java:15]: I would also rename the const FORM_FACTOR_MAP....

File: freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/scale/ScaleRequestBase.java [hunk 2]
```diff
@@ -29,19 +29,19 @@ public void setEnvironmentCrn(String environmentCrn) {
...
+    public void setTargetFormFactor(AvailabilityType targetAvailabilityType) {
+        this.targetAvailabilityType = targetAvailabilityType;
     }
 
     @Override
     public String toString() {
         return "ScaleRequest{" +
                 "environmentCrn='" + environmentCrn + '\'' +
-                ", targetFormFactor=" + targetFormFactor +
+                ", targetFormFactor=" + targetAvailabilityType +
```
Comment (gergopapi2) [inline freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/scale/ScaleRequestBase.java:44]: also here I would rename targetFormFactor=

File: freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/scale/ScaleResponseBase.java [hunk 3]
```diff
@@ -29,28 +29,28 @@ public void setOperationId(String operationId) {
         this.operationId = operationId;
     }
 
-    public FormFactor getOriginalFormFactor() {
-        return originalFormFactor;
+    public AvailabilityType getOriginalFormFactor() {
```
Comment (gergopapi2) [inline freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/scale/ScaleResponseBase.java:32]: I would rename the method to getOriginalAvailabilityType

File: freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/scale/ScaleResponseBase.java [hunk 4]
```diff
@@ -29,28 +29,28 @@ public void setOperationId(String operationId) {
...
+        return originalAvailabilityType;
     }
 
-    public void setOriginalFormFactor(FormFactor originalFormFactor) {
-        this.originalFormFactor = originalFormFactor;
+    public void setOriginalFormFactor(AvailabilityType originalAvailabilityType) {
```
Comment (gergopapi2) [inline freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/scale/ScaleResponseBase.java:36]: I would rename the method to setOriginalAvailabilityType

File: freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/scale/ScaleResponseBase.java [hunk 5]
```diff
@@ -29,28 +29,28 @@ public void setOperationId(String operationId) {
...
+        this.originalAvailabilityType = originalAvailabilityType;
     }
 
-    public FormFactor getTargetFormFactor() {
-        return targetFormFactor;
+    public AvailabilityType getTargetFormFactor() {
```
Comment (gergopapi2) [inline freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/scale/ScaleResponseBase.java:40]: Also, I would rename to get/setTargetAvailabilityType

File: freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/scale/ScaleResponseBase.java [hunk 6]
```diff
@@ -29,28 +29,28 @@ public void setOperationId(String operationId) {
...
+        this.targetAvailabilityType = targetAvailabilityType;
     }
 
     @Override
     public String toString() {
         return "ScaleResponse{" +
                 "flowIdentifier=" + flowIdentifier +
-                ", originalFormFactor=" + originalFormFactor +
-                ", targetFormFactor=" + targetFormFactor +
+                ", originalFormFactor=" + originalAvailabilityType +
```
Comment (gergopapi2) [inline freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/scale/ScaleResponseBase.java:52]: I would also rename the toString to original/targetAvailabilityType=

File: freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/scale/ScalingPath.java [hunk 7]
```diff
@@ -2,32 +2,32 @@
...
+    public AvailabilityType getTargetFormFactor() {
+        return targetAvailabilityType;
     }
 
     @Override
     public String toString() {
         return "ScalingPath{" +
-                "originalFormFactor=" + originalFormFactor +
-                ", targetFormFactor=" + targetFormFactor +
+                "originalFormFactor=" + originalAvailabilityType +
```
Comment (gergopapi2) [inline freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/scale/ScalingPath.java:29]: I would rename the toString lines to original/targetAvailabilityType=

File: freeipa-api/src/main/java/com/sequenceiq/service/api/doc/ModelDescriptions.java [hunk 8]
```diff
@@ -11,7 +11,7 @@
     public static final String CHILD_ENVIRONMENT_CRN = "CRN of the child environment";
     public static final String CLUSTER_CRN = "CRN of the cluster";
     public static final String INSTANCE_ID = "ID of the instance";
-    public static final String FORM_FACTOR = "Form factor of the resource reflecting the number of instances and the presence or absence of high availability";
+    public static final String AVAILABILITY_TYPE = "Availability type or form factor of the resource reflecting the number of instances and the presence or absence of high availability";
```
Comment (gergopapi2) [inline freeipa-api/src/main/java/com/sequenceiq/service/api/doc/ModelDescriptions.java:14]: I would delete the "or form factor"

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaScalingService.java [hunk 9]
```diff
@@ -54,26 +54,26 @@
...
+        AvailabilityType originalAvailabilityType = AvailabilityType.getByInstanceCount(allInstances.size());
+        logRequest(OperationType.UPSCALE, request, originalAvailabilityType);
+        validationService.validateStackForUpscale(allInstances, stack, new ScalingPath(originalAvailabilityType, request.getTargetFormFactor()));
+        return triggerUpscale(request, stack, originalAvailabilityType);
     }
 
-    private void logRequest(OperationType operationType, ScaleRequestBase request, FormFactor originalFormFactor) {
-        LOGGER.debug("{} request received with original form factor {} and request {}", operationType, originalFormFactor, request);
+    private void logRequest(OperationType operationType, ScaleRequestBase request, AvailabilityType originalAvailabilityType) {
+        LOGGER.debug("{} request received with original form factor {} and request {}", operationType, originalAvailabilityType, request);
```
Comment (gergopapi2) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaScalingService.java:64]: I would change the log message: form factor to availability type.

**Comment by pdarvasi** [comment]
Swagger failure is expected, can be ignored: 
image

File: freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/scale/ScalingPath.java [hunk 10]
```diff
@@ -2,32 +2,32 @@
...
-        this.originalFormFactor = originalFormFactor;
-        this.targetFormFactor = targetFormFactor;
+    public ScalingPath(AvailabilityType originalAvailabilityType, AvailabilityType targetAvailabilityType) {
+        this.originalAvailabilityType = originalAvailabilityType;
+        this.targetAvailabilityType = targetAvailabilityType;
     }
 
-    public FormFactor getOriginalFormFactor() {
-        return originalFormFactor;
+    public AvailabilityType getOriginalFormFactor() {
```
Comment (gergopapi2) [inline freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/scale/ScalingPath.java:18]: I would rename it to getOriginalAvailabilityType()

File: freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/scale/ScalingPath.java [hunk 11]
```diff
@@ -2,32 +2,32 @@
...
+        return originalAvailabilityType;
     }
 
-    public FormFactor getTargetFormFactor() {
-        return targetFormFactor;
+    public AvailabilityType getTargetFormFactor() {
```
Comment (gergopapi2) [inline freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/scale/ScalingPath.java:22]: Please rename to getTargetAvailabilityType()

**Review by gergopapi2 — CHANGES_REQUESTED** [review]

**Pull request merged**